network.c

```diff
@@ -1061,17 +1059,76 @@ struct iio_context * network_create_context(const char *host)
 		return NULL;
 	}
 #endif
+	/* ipv4 addresses are simply ip:port, e.g. 192.168.1.67:80
+	 * Because IPv6 addresses contain colons, and URLs use colons to separate
+	 * the host from the port number, RFC2732 (Format for Literal IPv6
+	 * Addresses in URL's) specifies that an IPv6 address used as the
+	 * host-part of a URL (our in our case the URI) should be enclosed
+	 * in square brackets, e.g.
+	 * [2001:db8:4006:812::200e] or [2001:db8:4006:812::200e]:8080
+	 * A compressed form allows one string of 0s to be replaced by ‘::'
+	 * e.g.: [FF01:0:0:0:0:0:0:1A2] can be represented by [FF01::1A2]
+	 *
+	 * RFC1123 (Requirements for Internet Hosts) requires valid hostnames to
+	 * only contain the ASCII letters 'a' through 'z' (in a case-insensitive
+	 * manner), the digits '0' through '9', and the hyphen ('-').
+	 *
+	 * We need to distinguish between:
+	 *   - ipv4       192.168.1.67                       Default Port
+	 *   - ipv4:port  192.168.1.67:50000                 Custom Port
+	 *   - ipv6       2001:db8:4006:812::200e            Default Port
+	 *   - [ipv6]     [2001:db8:4006:812::200e]          Default Port
```

Comment:
I'd be in favor of dropping support for IPv6 in brackets without the port specified, I never saw this used in the wild, and that would make the code below a bit smaller.

Reply:
Ok - dropped.